enterprise/payara.tooling/src/org/netbeans/modules/payara/tooling/data/PayaraVersion.java

```diff
@@ -18,354 +18,203 @@
-    static final String PF_4_1_2_174_STR_NEXT[] = {"4.1.2.174.0"};
-    /**  A <code>String</code> representation of PF_4_1_2_181 value. */
-    static final String PF_4_1_2_181_STR = "4.1.2.181";
-    /** Additional <code>String</code> representations of 4.1.2.181 value. */
-    static final String PF_4_1_2_181_STR_NEXT[] = {"4.1.2.181.0"};
-    /**  A <code>String</code> representation of PF_5_181 value. */
-    static final String PF_5_181_STR = "5.181";
-    /** Additional <code>String</code> representations of 5.181 value. */
-    static final String PF_5_181_STR_NEXT[] = {"5.181.0"};
-    /**  A <code>String</code> representation of PF_5_182 value. */
-    static final String PF_5_182_STR = "5.182";
-    /** Additional <code>String</code> representations of 5.182 value. */
-    static final String PF_5_182_STR_NEXT[] = {"5.182.0"};
-    /**  A <code>String</code> representation of PF_5_183 value. */
-    static final String PF_5_183_STR = "5.183";
-    /** Additional <code>String</code> representations of 5.183 value. */
-    static final String PF_5_183_STR_NEXT[] = {"5.183.0"};
-    /**  A <code>String</code> representation of PF_5_184 value. */
-    static final String PF_5_184_STR = "5.184";
-    /** Additional <code>String</code> representations of 5.184 value. */
-    static final String PF_5_184_STR_NEXT[] = {"5.184.0"};
-    /**  A <code>String</code> representation of PF_5_191 value. */
-    static final String PF_5_191_STR = "5.191";
-    /** Additional <code>String</code> representations of 5.191 value. */
-    static final String PF_5_191_STR_NEXT[] = {"5.191.0"};
-    /**  A <code>String</code> representation of PF_5_192 value. */
-    static final String PF_5_192_STR = "5.192";
-    /** Additional <code>String</code> representations of 5.192 value. */
-    static final String PF_5_192_STR_NEXT[] = {"5.192.0"};
-    /**  A <code>String</code> representation of PF_5_193 value. */
-    static final String PF_5_193_STR = "5.193";
-    /** Additional <code>String</code> representations of 5.193 value. */
-    static final String PF_5_193_STR_NEXT[] = {"5.193.0"};
-    /**  A <code>String</code> representation of PF_5_194 value. */
-    static final String PF_5_194_STR = "5.194";
-    /** Additional <code>String</code> representations of 5.194 value. */
-    static final String PF_5_194_STR_NEXT[] = {"5.194.0"};
-    /**  A <code>String</code> representation of PF_5_201 value. */
-    static final String PF_5_201_STR = "5.201";
-    /** Additional <code>String</code> representations of 5.201 value. */
-    static final String PF_5_201_STR_NEXT[] = {"5.201.0"};
-    /**  A <code>String</code> representation of PF_5_202 value. */
-    static final String PF_5_202_STR = "5.202";
-    /** Additional <code>String</code> representations of 5.202 value. */
-    static final String PF_5_202_STR_NEXT[] = {"5.202.0"};
-
+public class PayaraVersion implements Comparable<PayaraVersion> {
```

Comment:
To keep compatibility one would have to:
* keep the `public enum PayaraVersion`
* make it implement `Comparable`
* Introduce new (package private) `class PayaraVersionImproved`
* extract common functionality into `interface PayaraVersionApi`
* make both `enum` and "improved" `class` implement the API interface